include/jemalloc/internal/hpa_opts.h

```diff
@@ -59,6 +104,43 @@ struct hpa_shard_opts_s {
 	 * Maximum number of hugepages to purge on each purging attempt.
 	 */
 	ssize_t experimental_max_purge_nhp;
+
+	/*
+	 * When the number of inactive bytes in a hugepage is >=
+	 * purge_threshold, the page is purgable.  Setting this to HUGEPAGE
+	 * would only purge empty pages if using hugify_style_eager or
+	 * hugify_style_eager_and_lazy is true and the purges would be exactly
+	 * HUGEPAGE bytes.  Depending on your kernel settings, this may result
+	 * in better performance.  When purge_threshold is < HUGEPAGE and
+	 * hugify style is hugify_style_lazy, empty pages will be purgable even
+	 * when below threshold.
+	 *
+	 * Please note, when threshold is reached, we will purge all the dirty
+	 * bytes, and not just up to the threshold.  If this value is 1, then
+	 * all the pages are purgable.  We treat purgability constraint for
+	 * purge_threshold as stronger than dirty_mult, iow, if no page meets
+	 * purge_threshold, we will not purge even if we are above dirty_mult.
+	 */
+	size_t purge_threshold;
+
+	/*
+	 * Minimum number of ms that needs to elapse between HP page becoming
+	 * eligible for purging and actually getting purged.  Setting this to
+	 * a larger number would give better chance of reusing that memory.
+	 * Setting it to 0 means that page is eligible for purging as soon as
+	 * it meets the purge_threshold.  The clock resets when purgability of
+	 * the page changes (page goes from being non-purgable to purgable).
+	 * When using some of the eager styles, you probably want to allow for
+	 * some delay, to avoid purging the page too quickly and give it time
+	 * to be used.
+	 */
+	uint64_t min_purge_delay_ms;
+
+	/*
+	 * Style of madvise (see comment at hpa_hugify_style_t for
+	 * options).
+	 */
```

Comment:
This isn't strictly about `madvise(2)` so maybe we can explain this in a different way?